src/Jobs/ReorderNestedModuleItems.php

```diff
@@ -2,31 +2,37 @@
 
 namespace A17\Twill\Jobs;
 
-
+use A17\Twill\Models\Model;
+use A17\Twill\Models\User;
 use Illuminate\Bus\Queueable;
 use Illuminate\Contracts\Queue\ShouldQueue;
 use Illuminate\Foundation\Bus\Dispatchable;
 use Illuminate\Queue\InteractsWithQueue;
 use Illuminate\Support\Facades\DB;
-use A17\Twill\Models\Model;
 
 class ReorderNestedModuleItems implements ShouldQueue
 {
     use Dispatchable, InteractsWithQueue, Queueable;
 
     protected $modelClass;
     protected $ids;
+    protected $user;
 
-    public function __construct(Model $model, array $ids)
+    public function __construct(Model $model, array $ids, User $user)
     {
         $this->modelClass = get_class($model);
         $this->ids = $ids;
+        $this->user = $user;
     }
 
     public function handle()
     {
         DB::transaction(function () {
-            app($this->modelClass)->saveTreeFromIds($this->ids);
+            $savedModels = app($this->modelClass)->saveTreeFromIds($this->ids);
+            // Log the reordered models
+            foreach ($savedModels as $model) {
+                activity()->performedOn($model)->causedBy($this->user)->log('reordered');
```

Comment:
Wouldnt this cause a huge spam if you would move an item from pos 100 to 1? Since all other items are being moved with this as well?

Reply:
Yes, that is the problem, it should just log the initiator, but I cannot find a simple way do this